### PR TITLE
[TS] Use a one file model for typing custom commands

### PR DESCRIPTION
**Pull request by elevatebart**

### CODE REVIEW

**Comment by elevatebart** [comment]
I could as well transform this documentation to have the merged files as the main explanation.
It is a change, but would really help our users I believe.

File: content/guides/tooling/typescript-support.md [hunk 1]
```diff
@@ -95,9 +95,8 @@ If your specs files are in TypeScript, you should include the TypeScript definit
 
 Even if your project is JavaScript only, the JavaScript specs can know about the new command by referencing the file using the special triple slash `reference path` comment.
 
-```javascript
+```typescript
```
Comment (jennifer-shehane) [inline content/guides/tooling/typescript-support.md:98]: This area appears to be addressing JS specifically, which is why the example had the triple slash references, but I do think this is likely confusing people. I would remove this paragraph "Even if your project is Javascript only....". The doc is about TypeScript after all.

**Review by jennifer-shehane — CHANGES_REQUESTED** [review]
There's this section also in the Custom commands section:

>If your specs files are in TypeScript, you should include the TypeScript definition file, `cypress/support/index.d.ts`, with the rest of the source files.

This makes it sound like I have to do an extra step, like an 'include'. But it seems like my TS is working for my custom command without doing anything else? I would remove this line if it's no longer necessary.

**Comment by elevatebart** [comment]
Thank you @jennifer-shehane,

I updated the docs to include your remarks. 
I think I should as well add a link to here from the [custom commands docs](https://docs.cypress.io/api/cypress-api/custom-commands#See-also)

**Comment by elevatebart** [comment]
It is now linked !

File: content/api/cypress-api/custom-commands.md [hunk 2]
```diff
@@ -783,6 +783,7 @@ You can describe the method signature for your custom command, allowing IntelliS
 
 ## See also
 
+- See how to add Typescript support in the documentation [Types for custom commands](/guides/tooling/typescript-support#Types-for-custom-commands)
```
Comment (jennifer-shehane) [inline content/api/cypress-api/custom-commands.md:786]:
```suggestion
- See how to add [TypeScript support for custom commands](/guides/tooling/typescript-support#Types-for-custom-commands)
```

**Review by jennifer-shehane — APPROVED** [review]

**Pull request merged**